### PR TITLE
Fix for issue #5631

### PR DESCRIPTION
**Pull request by emeraldinspirations**
- Add " ( and ) as valid characters in author name
- Add relevant unit tests
#5631

### CODE REVIEW

**Comment by alcohol** [comment]
LGTM

File: src/Composer/Command/InitCommand.php [hunk 1]
```diff
@@ -320,7 +320,7 @@ function ($value) use ($self, $minimumStability) {
      */
     public function parseAuthorString($author)
     {
-        if (preg_match('/^(?P<name>[- \.,\p{L}\p{N}\'’]+) <(?P<email>.+?)>$/u', $author, $match)) {
+        if (preg_match('/^(?P<name>[- \.,\p{L}\p{N}\'’\"\(\)]+) <(?P<email>.+?)>$/u', $author, $match)) {
```
Comment (avant1) [inline src/Composer/Command/InitCommand.php]: You are not required to escape double quote and parentheses in "character class" (characters sequence inside square brackets). The same is actual for existing dot.
The results probably will be the same, but pattern will look less complicated.

Comment (alcohol) [inline src/Composer/Command/InitCommand.php]: That is correct. It could be simplified to:

```
^(?P<name>[- .,\p{L}\p{N}'’"()]+) <(?P<email>.+?)>$
```

**Comment by emeraldinspirations** [comment]
@alcohol sorry, new to github.  What does LGTM mean?  P.S. is there some sort of abbreviation key somewhere?

**Comment by alcohol** [comment]
Looks good to me. :-)

**Comment by curry684** [comment]
@emeraldinspirations http://acronymfinder.com/Looks-Good-To-Me-(LGTM).html hehe

**Comment by Seldaek** [comment]
Thanks, merged in 1.2 so it won't show merged here, but it's in the composer snapshots now and will be in the next 1.2.x release.

**Pull request closed without merging**